### PR TITLE
Supports maliput_osm as backend for delphyne_gazoo demo.

### PR DESCRIPTION
**Pull request by francocipollone**
# 🎉 New feature

Relies on https://github.com/maliput/delphyne/pull/858

## Summary
Supports using maliput_osm backend for the `delphyne_gazoo` demo.

https://user-images.githubusercontent.com/53065142/201933179-26f462e1-f2f2-4ce5-b8fa-d23773401130.mp4

## Test it

```
delphyne_gazoo -m maliput_osm
```

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if it affects the public API)

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: demos/gazoo.py [hunk 1]
```diff
@@ -66,21 +100,44 @@ def parse_arguments():
...
+    filename = obtain_circuit_filepath(backend)
+
     # The road geometry
-    scenario_subtree = delphyne.behaviours.roads.Multilane(
-        file_path=filename,
-        name="circuit",
-    )
+    features = delphyne.roads.ObjFeatures()
+    features.draw_elevation_bounds = False
+    if(backend == "maliput_multilane"):
```
Comment (agalbachicar) [inline demos/gazoo.py:118]: Technical debt: we must change string based enums to be https://docs.python.org/3/library/enum.html

File: demos/gazoo.py [hunk 2]
```diff
@@ -51,6 +51,40 @@
 # Supporting Classes & Methods
 ##############################################################################
 
+def get_maliput_osm_circuit():
```
Comment (agalbachicar) [inline demos/gazoo.py:54]: Can't we make a POD with configurations based on the backend? We can absorb all if-elseif-else cases and collapse them into just one after parsing the selected backend?

This way code will have way less branching and will be much simpler to read and maintain in the future.

**Review by agalbachicar — CHANGES_REQUESTED** [review]
PTAL

File: demos/gazoo.py (hunk 2 above)
Comment (francocipollone) [reply demos/gazoo.py:63]: Let me know if now it is a bit cleaner.

**Review by agalbachicar — APPROVED** [review]
LGTM

**Pull request merged**